### PR TITLE
ESQL: Document `date` instead of `datetime`

### LINKED ISSUE #111650: ESQL: Align documentation and the return type for datetime/date type
### Description

It would be good to align the types mentioned in the documentation to what we are returning in the ES|QL reponses. Either updating the documentation to use `date` instead of `datetime` or change the return type of the response to `datetime` . Currently, the documentation uses a mix of both which is a bit confusing.

For example, let's say for following documentation of [date_diff](https://www.elastic.co/guide/en/elasticsearch/reference/current/esql-functions-operators.html#esql-date_diff) that uses  to_datetime, we are mixing datetime (supported type table above) and :date (table below).

image

image

Supported field types also mention `date`

image

Additionally, it would be good to clarify what `date_period` and `time_duration` are and some examples in the ES|QL documentation.

### PR DESCRIPTION
**Pull request by nik9000**
This changes the generated types tables in the docs to say `date` instead of `datetime`. That's the name of the field in Elasticsearch so it's a lot less confusing to call it that.

Closes #111650

### CODE REVIEW

**Comment by nik9000** [comment]
@drewdaemon would y'all prefer `date` over `datetime`?

I believe we have the `datetime` name to differentiate from a `date` without a time - like you can do in some SQL implementations. But ESQL doesn't have that and doesn't plan to. And `date` is what Elasticsearch calls the field.

**Comment by drewdaemon** [comment]
@nik9000 yes, our take is that if there's no reason to differentiate, let's keep it simple. On our end, we interpret `date` and `datetime` in the function definitions as identical.

**Comment by drewdaemon** [comment]
So I guess the main point is to be consistent. Beyond that, `date` seems to me like the best candidate because that is what the field type is called.

File: docs/reference/esql/functions/kibana/docs/to_datetime.md [hunk 1]
```diff
@@ -11,3 +11,4 @@ To convert dates in other formats, use <<esql-date_parse>>.
 ROW string = ["1953-09-02T00:00:00.000Z", "1964-06-02T00:00:00.000Z", "1964-06-02 00:00:00"]
 | EVAL datetime = TO_DATETIME(string)
 ```
+Note: Note that when converting from nanosecond resolution to millisecond resolution with this function, the nanosecond date istruncated, not rounded.
```
Comment (marciw) [inline docs/reference/esql/functions/kibana/docs/to_datetime.md:14]:
```suggestion
Note: Note that when converting from nanosecond resolution to millisecond resolution with this function, the nanosecond date is truncated, not rounded.
```

File: docs/reference/esql/functions/kibana/definition/to_datetime.json [hunk 2]
```diff
@@ -3,6 +3,7 @@
   "type" : "eval",
   "name" : "to_datetime",
   "description" : "Converts an input value to a date value.\nA string will only be successfully converted if it's respecting the format `yyyy-MM-dd'T'HH:mm:ss.SSS'Z'`.\nTo convert dates in other formats, use <<esql-date_parse>>.",
+  "note" : "Note that when converting from nanosecond resolution to millisecond resolution with this function, the nanosecond date istruncated, not rounded.",
```
Comment (marciw) [inline docs/reference/esql/functions/kibana/definition/to_datetime.json:6]:
```suggestion
  "note" : "Note that when converting from nanosecond resolution to millisecond resolution with this function, the nanosecond date is truncated, not rounded.",
```

File: docs/reference/esql/functions/description/to_datetime.asciidoc [hunk 3]
```diff
@@ -3,3 +3,5 @@
 *Description*
 
 Converts an input value to a date value. A string will only be successfully converted if it's respecting the format `yyyy-MM-dd'T'HH:mm:ss.SSS'Z'`. To convert dates in other formats, use <<esql-date_parse>>.
+
+NOTE: Note that when converting from nanosecond resolution to millisecond resolution with this function, the nanosecond date istruncated, not rounded.
```
Comment (marciw) [inline docs/reference/esql/functions/description/to_datetime.asciidoc:7]:
```suggestion
NOTE: Note that when converting from nanosecond resolution to millisecond resolution with this function, the nanosecond date is truncated, not rounded.
```

**Review by marciw — APPROVED** [review]
Thank you @nik9000 -- clearer and simpler this way! I flagged a few instances of a li'l typo (`istruncated`)

File: docs/reference/esql/functions/kibana/docs/to_datetime.md (hunk 1 above)
Comment (nik9000) [reply docs/reference/esql/functions/kibana/docs/to_datetime.md:14]: Will fix!

**Comment by nik9000** [comment]
> So I guess the main point is to be consistent. Beyond that, `date` seems to me like the best candidate because that is what the field type is called.

:+1:

I'll update this to update your json too.

**Comment by nik9000** [comment]
I've marked this as auto-merge. I'll have to backport it by hand because we'll need to regenerate all of the docs changes. That's fine.

**Comment by drewdaemon** [comment]
This is awesome. Thanks @nik9000

**Comment by nik9000** [comment]
Backported in #112034.

**Pull request merged**